libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1109 @@
+        sub_cmd : static_cast<uint8_t>(DeviceSubCmd::Firmware_Info),
+        data : NULL
+    };
+    
+    // send packet to gimbal
+    return send_packet(frame_to_send);
+}
+
+// send request for models, gimbal will respond models after recieve this message
+bool AP_Mount_Tusuav::request_models()
+{
+    // fill in packet
+    FrameType frame_to_send{
+        .flag={0x28},
+        data_len : 0,
+        cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+        sub_cmd : static_cast<uint8_t>(GimbalSubCmd::GimbalSubCmd_GimbalSn),
+        data : NULL
+    };
+    
+    // send packet to gimbal
+    return send_packet(frame_to_send);
+}
+
+
+// set gimbal's lock vs follow mode
+// lock should be true if gimbal should maintain an earth-frame target
+// lock is false to follow / maintain a body-frame target
+bool AP_Mount_Tusuav::set_lock(bool lock)
+{
+    // do not send if lock mode has already been sent recently
+    if (_last_lock == lock) {
+        return true;
+    }
+
+    PodCtrlUnion gimbal_ctrl_settings;
+    gimbal_ctrl_settings.all = _gimbal_ctrl_settings.all;
+
+    gimbal_ctrl_settings.bits.lock_ef = lock;
+
+    // fill in packet
+    FrameType frame_to_send{
+        .flag={0x00},
+        data_len : sizeof(gimbal_ctrl_settings),
+        cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+        sub_cmd : static_cast<uint8_t>(GimbalSubCmd::GimbalSubCmd_PodCtrlReg),
+        data : &gimbal_ctrl_settings.all
+    };
+    
+    // send packet to gimbal
+    return send_packet(frame_to_send);
+}
+
+// send target pitch and yaw angles to gimbal
+// fill the settings for choosing an earth-frame / body-frame before calling send_target_angles_rates function
+bool AP_Mount_Tusuav::send_target_angles_rates(MessageGimbalAngleRates mode)
+{
+    // gimbal does not support lock in angle control mode
+    if (!set_lock(false)) {
+        return false;
+    }
+
+    // fill in packet
+    FrameType frame_to_send{
+        .flag={0},
+        data_len : sizeof(mode),
+        cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+        sub_cmd : static_cast<uint8_t>(GimbalSubCmd::GimbalSubCmd_PlaneControlPara),
+        data : (uint8_t*)&mode
+    };
+
+    // send targets to gimbal
+    return send_packet(frame_to_send);
+}
+
+// send reference of the joystick rocker for moving the gimbal or tracking window
+// when gimbal runs in tracking mode this command will move the tracking window for selecting the target
+// when in other mode(except angle mode) this command will set the rates of the yaw axis and pitch axis(±500 map to max/min rate speed)
+bool AP_Mount_Tusuav::send_stick_pitch_yaw(int16_t pitch,int16_t yaw)
+{
+    // gimbal does not support lock in angle control mode
+    if (!set_lock(false)) {
+        return false;
+    }
+
+    if (pitch > 500) {
+		pitch = 500;
+	}
+	else if (pitch < -500) {
+		pitch = -500;
+	}
+
+    if (yaw > 500) {
+    	yaw = 500;
+	}
+	else if (yaw < -500) {
+		yaw = -500;
+	}
```

Comment:
```suggestion
		pitch = 500;
	}
	else if (pitch < -500) {
		pitch = -500;
	}

    if (yaw > 500) {
    	yaw = 500;
	}
	else if (yaw < -500) {
		yaw = -500;
	}
```


indenting / whitespace